src/screens/FreeCommunityScreen.jsx

```diff
@@ -19,149 +19,165 @@ import Constants from 'expo-constants';
 import i18n from 'i18n-js';
 import * as Network from 'expo-network';
 import { Snackbar } from 'react-native-paper';
+import { object } from 'prop-types';
 
 class FreeCommunityScreen extends Component {
-	state = {
-		generations: [],
-	};
-
-	async componentDidMount() {
-		const status = await Network.getNetworkStateAsync()
-		if (status.isConnected === true) {
-			axios
-				.get(`${i18n.locale}/api/v1/generations?fields=all&key=${Constants.manifest.extra.secretKey}`)
-				.then((res) => {
-					const fetchedGenerations = res.data.result.map(entry => {
-						return {
-							...entry,
-							data: entry.courses
-						}
-					})
-					this.setState({ generations: fetchedGenerations });
-				}).catch(err => {
-					this.setState({ snackMsg: i18n.t('GENERAL.ERROR'), visible: true, loading: false })
-				});
-		} else {
-			this.setState({ snackMsg: i18n.t('GENERAL.NO_INTERNET'), visible: true, loading: false })
-		}
-
-	}
-
-	render() {
-		let TouchableComp = TouchableOpacity;
-		if (Platform.OS === 'android' && Platform.Version >= 21) {
-			TouchableComp = TouchableNativeFeedback;
-		}
-		return (
-			<SafeAreaView style={styles.container}>
-				{this.state.generations.length > 0 ?
-					<SectionList
-						sections={this.state.generations}
-						renderItem={({ item }) => <Course key={item.courseId} title={item.name} onSelect={() => this.props.navigation.navigate('CourseDetail', { courseId: item.courseId })} />}
-						renderSectionHeader={({ section: { name, generationId } }) => (
-							<TouchableComp key={generationId} onPress={(section) => {
-								console.log(section)
-								this.props.navigation.navigate('GenerationDetail', { generationId: generationId })
-							}}>
-								<View style={styles.sectionHeaderContainer}>
-									<Text style={styles.header}>{name}</Text>
-									<Ionicons name='ios-help-circle-outline' size={23} color={Colors.primaryColor} />
-								</View>
-
-							</TouchableComp>
-						)}
-					/>
-					:
-					<View>
-						<ActivityIndicator size="large" color={Colors.primaryColor} />
-					</View>
-				}
-				<Snackbar visible={this.state.visible} onDismiss={() => this.setState({ visible: false })} style={styles.snackError}>
-					{this.state.snackMsg}
-				</Snackbar>
-
-			</SafeAreaView>
-		);
-	}
+  state = {
+    generations: [],
+  };
+
+  async componentDidMount() {
+    const status = await Network.getNetworkStateAsync();
+    if (status.isConnected === true) {
+      axios
+        .get(`${i18n.locale}/api/v1/courses?fields=all&key=${Constants.manifest.extra.secretKey}`)
+        .then((res) => {
+          var data = [];
+          for (var a of res.data.result) {
+            let i = a.generationId == null ? 0 : a.generationId;
+            data[i] = data[i] || { data: [], name: '', generationId: '' };
+            data[i].name = a.generationName || 'Sin generacion';
```

Comment:
'Sin generacion' is a string literal, we need support for i18n,, eg: en, es, pt, de not only es.